### PR TITLE
fix: [optioncore] Fixed that shortcut keys could not be cancelled

### PR DESCRIPTION
**Pull request by fuyou158**
Log: Fixed that shortcut keys could not be cancelled, added right-click menu to cancel shortcut keys

Bug: https://pms.uniontech.com/bug-view-246835.html

### CODE REVIEW

File: src/plugins/option/optioncore/mainframe/shortcutedit.cpp [hunk 1]
```diff
@@ -0,0 +1,18 @@
+// SPDX-FileCopyrightText: 2023 UnionTech Software Technology Co., Ltd.
```
Comment (LiHua000) [inline src/plugins/option/optioncore/mainframe/shortcutedit.cpp:1]: 2024

File: src/plugins/option/optioncore/mainframe/shortcutedit.cpp [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+ShortCutEdit::ShortCutEdit(QWidget *parent)
+    : DKeySequenceEdit(parent)
+{
+    menu = new DMenu(this);
+    QAction *clearAction = menu->addAction(tr("Clear"));
+    connect(clearAction, &QAction::triggered, this, &ShortCutEdit::clearText);
+}
+
+void ShortCutEdit::clearText() {
```
Comment (Kakueeen) [inline src/plugins/option/optioncore/mainframe/shortcutedit.cpp:15]: 编码规范，大括号应该在函数名下面

File: src/plugins/option/optioncore/mainframe/shortcutedit.h [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    }
+
+signals:
+    void clearShortCut();
+
+private slots:
+    void clearText();
+
+private:
+    DMenu *menu;
```
Comment (Kakueeen) [inline src/plugins/option/optioncore/mainframe/shortcutedit.h:31]: 赋初值

File: src/plugins/option/optioncore/mainframe/shortcutedit.cpp (hunk 1 above)
Comment (fuyou158) [reply src/plugins/option/optioncore/mainframe/shortcutedit.cpp]: 已修改

File: src/plugins/option/optioncore/mainframe/shortcutedit.h [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+public:
+    ShortCutEdit(QWidget *parent = nullptr);
+
+protected:
+    void contextMenuEvent(QContextMenuEvent *event) override {
+        menu->exec(event->globalPos());
+    }
+
+signals:
+    void clearShortCut();
```
Comment (Kakueeen) [inline src/plugins/option/optioncore/mainframe/shortcutedit.h:25]: 函数名跟实际表达的含义不一致，shortcutCleared

**Review by Kakueeen — APPROVED** [review]

**Review by LiHua000 — APPROVED** [review]

**Pull request merged**